src/arrayMethodSort.js

```diff
@@ -4,8 +4,43 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let len = this.length;
+    let swapped;
+
+    do {
+      swapped = false;
+
+      for (let i = 0; i < len - 1; i++) {
+        if (compareFunction) {
+          if (compareFunction(this[i], this[i + 1]) > 0) {
+            const temp = this[i];
+
+            this[i] = this[i + 1];
+            this[i + 1] = temp;
+
+            swapped = true;
+          } else if (
+            compareFunction(this[i], this[i + 1]) < 0 ||
+            compareFunction(this[i], this[i + 1]).isNan()
+          ) {
+            swapped = true;
+          }
```

Comment:
The `else if` condition will always set `swapped` to `true` when the comparison function returns a value less than 0, which is incorrect. The purpose of the `swapped` flag is to determine if any swaps have been made in the current iteration. If no swaps are made, the array is sorted. You should remove this `else if` block, as it serves no purpose in the logic of the sort.